.gitignore

```diff
@@ -1,42 +1,33 @@
-# .
-/eclipse-build
-/maven-build
-/classes
-build
-/idea-build
-dist
-lib
-test-lib
-/*~
-/build.properties
-/.idea
-lucene/**/*.iml
-parent.iml
-*.ipr
-*.iws
-/*.iml
-.project
-.classpath
-.settings
-/.caches
-/prj.el
-bin
-/bin
-/bin.*
-pom.xml
-/nbproject
-/nb-build
-.pydevproject
-__pycache__
-/dev-tools/scripts/scripts.iml
-.DS_Store
+### At the root
+
+/.idea/
+# Ignore the generated local settings file.
+/gradle.properties
+
+/prj.el # TODO?
+
+# Eclipse
+/.project
+/.classpath
+/.settings/
+
+### Any dir
 
-build/
 .gradle/
-.idea/
+build/
 
-# Ignore the generated local settings file.
-gradle.properties
+# macOS
+.DS_Store
+
+# Eclipse Gradle oddity
+**/bin/default/
```

Comment:
Move this to the other Eclipse section?

Reply:
What you refer to as "the other Eclipse section" is in a higher level "at the root", and this rule here is not at the root, it's at any directory.  That said, I suppose I should toss out that "at the root" label as it's better to put all these eclipse rules together.

Reply:
I don't use Eclipse so this doesn't really affect me much, but I think it's easier to have contiguous sections I can ignore instead of it being piecemeal throughout the file.